artifactory/commands/transferinstall/installplugin.go

```diff
@@ -0,0 +1,347 @@
+	NotValidDestinationErr = errors.Errorf("Can't find plugin directory with the provided information, this command must run on the Artifactory server.")
+	minVerErr              = errorutils.CheckErrorf("This operation requires Artifactory version %s or higher", minArtifactoryVersion)
+)
+
+type PluginFileItem []string
+type PluginFiles []PluginFileItem
+
+// Get the name componenet of the item
+func (f *PluginFileItem) Name() string {
+	size := len(*f)
+	if size == 0 {
+		return ""
+	}
+	return (*f)[size-1]
+}
+
+// Get the directory list of the item, ignore empty entries
+func (f *PluginFileItem) Dirs() *PluginFileItem {
+	dirs := PluginFileItem{}
+	for i := 0; i < len(*f)-1; i++ {
+		dir := (*f)[i]
+		if dir != "" {
+			dirs = append(dirs, dir)
+		}
+	}
+	return &dirs
+}
+
+// Split and get the componenets of the item
+func (f *PluginFileItem) SplitNameAndDirs() (string, *PluginFileItem) {
+	return f.Name(), f.Dirs()
+}
+
+// Convert the item to URL representation, ignore empty entries, adding prefix tokens as provided
+func (f *PluginFileItem) toURL(previousTokens ...string) string {
+	return toURL(toURL(previousTokens...), toURL(*f...))
+}
+
+// Convert the item to path representation, ignore empty entries, adding prefix tokens as provided
+func (f *PluginFileItem) toPath(previousTokens ...string) string {
+	return filepath.Join(filepath.Join(previousTokens...), filepath.Join(*f.Dirs()...), f.Name())
+}
+
+// Convert tokens into URL representation, ignore empty ("") entries
+func toURL(tokens ...string) string {
+	url := ""
+	for _, token := range tokens {
+		if token == "" {
+			continue
+		}
+		if url != "" {
+			url += "/"
+		}
+		url += token
+	}
+	return url
+}
+
+// Holds all the plugin files and options of destinations to transfer them into
+type PluginTransferManager struct {
+	files        PluginFiles
+	destinations []PluginFileItem
+}
+
+// Create new file transfer manager for artifactory plugins
+func NewArtifactoryPluginTransferManager(bundle PluginFiles) *PluginTransferManager {
+	manager := &PluginTransferManager{
+		files:        bundle,
+		destinations: []PluginFileItem{},
+	}
+	// Add all the optional destinations for the plugin dir
+	manager.addDestination(OriginalDirPath)
+	manager.addDestination(V7DirPath)
+	return manager
+}
+
+// Add optional plugin directory location as destination
+func (ftm *PluginTransferManager) addDestination(directory PluginFileItem) {
+	ftm.destinations = append(ftm.destinations, directory)
+}
+
+// Search all the local target directories that the plugin directory can exist in base on a given root JFrog artifactory home directory
+// the first option that matched and the directory exists is returned as target
+func (ftm *PluginTransferManager) trySearchDestinationMatchFrom(rootDir string) (exists bool, target PluginFileItem, err error) {
+	if exists, err = fileutils.IsDirExists(rootDir, false); err != nil || !exists {
+		return
+	}
+	exists = false
```

Comment:
its not, because I use the same var before for checking if the root dir exists, if it does not it will return so it will always be true when getting to this line